tests/conftest.py

```diff
@@ -45,6 +57,27 @@
 # Test fixtures
 #
 
+@pytest.fixture(scope='session', autouse=True)
+def _check_git_capabilities(tmpdir_factory):
+    # Do checks for git behaviors. Right now this is limited to
+    # deciding whether or not 'git init --branch' is supported.
+    #
+    # We aren't using WestCommand._parse_git_version() here just to
+    # try to keep the conftest behavior independent of the code being
+    # tested.
+    global GIT_INIT_HAS_BRANCH
+
+    tmpdir = tmpdir_factory.mktemp("west-check-git-caps-tmpdir")
+
+    try:
+        subprocess.run([GIT, 'init', '--initial-branch', 'foo',
+                        os.fspath(tmpdir)],
+                       stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL,
+                       check=True)
+        GIT_INIT_HAS_BRANCH = True
+    except subprocess.CalledProcessError:
```

Comment:
I had a `CalledProcessError` for the same feature in the first version of #495 and you asked me to change it to `self.git_version_info >= (2, 28, 0)`. Is it because this is test code, can you elaborate?

Reply:
>  Is it because this is test code, can you elaborate?

Yep, see the comment:

```py
    # We aren't using WestCommand._parse_git_version() here just to
    # try to keep the conftest behavior independent of the code being
    # tested.
```